### PR TITLE
Article Content Scraped, AI Content Detection

### PR DESCRIPTION
**Pull request by laurenx1**
### Article Scraping 

- Uses **Selenium** and **BeautifulSoup** to scrape articles from the url provided by [newsdata.io](newsdata.io)
- runs python script server.py and scraper.py to do so
- cleans the text so that article is readable 
- includes newlines which indicates new paragraphs for good formatting
- must be hosted on virtual environment venv which includes library `flask`

### AI content detection

- Uses openAI's **RoBERTa** model to detect AI-generated text within the article hosted through **Hugging Face**
- This model outputs a "Real" and "Fake" percentage score (1 - Real = Fake and vice versa)

### Things to fix

- Annoying popup window with article url when scraping, must block
- content score recalculated each time article is opened, must save to Article model 
- poor formatting of Story, FeaturedStories

### CODE REVIEW

File: backend/index.js [hunk 1]
```diff
@@ -125,3 +127,77 @@ app.patch('/api/users/:id', async (req, res) => {
...
+            },
+        });
+        const articles = response.data.results || [];
+
+        for (const article of articles) {
+            await prisma.article.upsert({
+                where: { title: article.title },
+                update: {},
+                create: {
+                    title: article.title || 'No title available',
```
Comment (vnikithak) [inline backend/index.js:151]: Do you want to even display the articles that do not have a title?

File: frontend/src/components/FeaturedStories.jsx [hunk 2]
```diff
@@ -11,40 +11,100 @@ const FeaturedStories = ({ user }) => {
     useEffect(() => {
         const fetchTopStories = async () => {
             try {
-                const response = await axios.get(`https://newsdata.io/api/1/latest?`, {
-                    params: {
-                        apikey: apiKey,
-                        q: `breaking`,
```
Comment (vnikithak) [inline frontend/src/components/FeaturedStories.jsx:17]: Please add comments on what the parameters mean.

File: frontend/src/components/FeaturedStories.jsx [hunk 3]
```diff
@@ -11,40 +11,100 @@ const FeaturedStories = ({ user }) => {
...
-                setTopStories(response.data.results || []);
+                const response = await axios.get(import.meta.env.VITE_BACKEND_URL + '/api/articles');
+                setTopStories(response.data || []);
             } catch (error) {
-                console.log(apiKey);
                 console.error('Error fetching top stories:', error);
             }
         };
 
+        // Custom combination function
```
Comment (vnikithak) [inline frontend/src/components/FeaturedStories.jsx:21]: Please add some comments on what we're trying to achieve with this combination function.

**Review by vnikithak — COMMENTED** [review]
Overall the code looks good. Please try to make smaller PRs in the future and please add a video to the test plan to show what the change is.

File: backend/index.js (hunk 1 above)
Comment (laurenx1) [reply backend/index.js:151]: every article with newsdata has a title - just not always an author, description, or date published. I guess the logic statement here is unnecessary

File: frontend/src/components/FeaturedStories.jsx (hunk 3 above)
Comment (laurenx1) [reply frontend/src/components/FeaturedStories.jsx:21]: will be in next commit

**Pull request closed without merging**